### ISSUE TITLE
toHaveStyleRule didn't see changes to the styles by styled(Component)

### ISSUE THREAD

**Issue by RIP21** [styled-components/jest-styled-components#30]
I have this test:
```javascript
  it("styles got updated by styled-components", () => {
    //given
    const Styled = styled(Label)`
      background-color: seagreen;
    `;
    //when
    const tree = mount(<Styled>label</Styled>);
    //then
    expect(tree.find(Styled)).toHaveStyleRule("background-color", "seagreen");
  });
```
And I have that component under test:
```javascript
import React from "react";
import PT from "prop-types";
import styled, { css } from "styled-components";

const Label = ({ children, isRequired, ...rest }) =>
  children ? <label {...rest}>{children}{isRequired && <i>*</i>}</label> : null;

Label.propTypes = {
  /** Flag that will add red `*` in after the label */
  isRequired: PT.bool,
  /** Makes label to be inlined with input */
  horizontal: PT.bool,
  /** To override label styles when he is a part of other component */
  css: PT.array,
  className: PT.string
};

// language=SCSS prefix=dummy{ suffix=}
export const StyledLabel = styled(Label)`
  display: block;
  ${props => props.horizontal && css`
    display: inline;
  `}
  & i {
    color: red;
  }
  ${props => props.css}
`;

export default StyledLabel;

```

The result is weird:
```
 FAIL  src\components\atoms\Label\index.test.js
  ● <Label /> › styles got updated by styled-components
            "Expected iobhyf to have background-color matching seagreen"
            "But received, display: block;"
```
In the same time `tree.find(Label)`  gives
```
"Expected [object Object] to be a component from react-test-renderer, or a mounted enzyme component."
"But had an error, TypeError: Cannot read property 'generatedClassName' of null"
```
and `tree.find('label')` gives:
```
"Expected [object Object] to be a component from react-test-renderer, or a mounted enzyme component."
"But had an error, TypeError: Cannot read property '_renderedComponent' of undefined"
```
Snapshot looks like that:
```
exports[`<Label /> styles got updated by styled-components 1`] = `
.iobhyf {
  display: block;
}

.iobhyf i {
  color: red;
}

.hBRMHQ {
  background-color: seagreen;
}

<label
  className="sc-bwzfXH hBRMHQ sc-bdVaJa iobhyf"
>
  label
</label>
`;
```
I have `styled-components 2.0.1` `jest-styled-components 3.0.1` `enzyme 2.8.2`

**Comment by MicheleBertoli** [comment]
Thank you very much for opening the issue @RIP21.
It's fixed in v3.0.2, please upgrade the package.

**Comment by RIP21** [comment]
Yay! @MicheleBertoli thanks a lot :) Now it's working fine :)
